src/NUnitFramework/framework/Constraints/IConstraint.cs

```diff
@@ -61,6 +64,16 @@ public interface IConstraint : IResolveConstraint
         /// <returns>A ConstraintResult</returns>
         ConstraintResult ApplyTo<TActual>(ref TActual actual);
 
+        /// <summary>
+        /// Applies the constraint to a delegate that returns the task.
+        /// The default implementation simply evaluates the delegate and awaits the task
+        /// but derived classes may override it to provide for delayed processing.
+        /// </summary>
+        /// <typeparam name="TActual"></typeparam>
+        /// <param name="delTask"></param>
+        /// <returns></returns>
+        Task<ConstraintResult> ApplyToAsync<TActual>(Func<Task<TActual>> delTask);
+
```

Comment:
Could we add this to an `IAsyncConstraint` interface?
When resolving the constraint in `ThatAsync` you can then test on `IAsyncConstraint`?
If not, throw a `NotSupportedException`?

That way it would be a non-breaking change.

Reply:
That makes sense.
I've made IAsyncConstraint internal for now - in case we'd need to more task-related methods later.